cpu/sam0_common/periph/i2c.c

```diff
@@ -125,23 +125,16 @@ void i2c_init(i2c_t dev)
     /* Find and set baudrate. Read speed configuration. Set transfer
      * speed: SERCOM_I2CM_CTRLA_SPEED(0): Standard-mode (Sm) up to 100
      * kHz and Fast-mode (Fm) up to 400 kHz */
-    switch (i2c_config[dev].speed) {
-        case I2C_SPEED_NORMAL:
-        case I2C_SPEED_FAST:
-            bus(dev)->CTRLA.reg |= SERCOM_I2CM_CTRLA_SPEED(0);
-            break;
-        case I2C_SPEED_HIGH:
-            bus(dev)->CTRLA.reg |= SERCOM_I2CM_CTRLA_SPEED(2);
-            break;
-        default:
-            DEBUG("BAD BAUDRATE\n");
-            return;
+    if (i2c_config[dev].speed > I2C_SPEED_FAST) {
+        bus(dev)->CTRLA.reg |= SERCOM_I2CM_CTRLA_SPEED(2);
```

Comment:
![image](https://user-images.githubusercontent.com/1301112/84087335-24057400-a9ea-11ea-926f-d59a900e456f.png)

should better account for that too.